llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java

```diff
@@ -680,8 +678,8 @@ void tryScheduleUnderLock(final TaskWrapper taskWrapper) throws RejectedExecutio
       LOG.info("Attempting to execute {}", taskWrapper);
     }
     TaskRunnerCallable task = taskWrapper.getTaskRunnerCallable();
-    task.setWmCountersRunning();
     ListenableFuture<TaskRunner2Result> future = executorService.submit(task);
+    task.setWmCountersRunning();
```

Comment:
this is more a question than a comment. Am not very familiar with this part of the code, but submitting a runnable does not necessarily means it is running right ? for instance if the pool size is smaller than the number of runnable ? thanks